packages/components/src/internal/components/samples/SampleAssayDetail.tsx

```diff
@@ -78,38 +100,115 @@ const SampleAssayDetailButtons: FC<SampleAssayDetailButtonsProps> = props => {
+    const {
+        actions,
+        assayModel,
+        queryModels,
+        sampleModel,
+        tabOrder,
+        showImportBtn,
+        isSourceSampleAssayGrid,
+        showAliquotViewSelector,
+        onSampleAliquotTypeChange,
+        activeSampleAliquotType,
+        emptyAssayDefDisplay,
+        emptyAssayResultDisplay,
+        emptyAliquotViewMsg,
+        emptySampleViewMsg,
+        onTabChange,
+        activeTabId
+    } = props;
     const allModels = Object.values(queryModels);
     const allLoaded = allModels.every(model => !model.isLoading);
 
     useEffect(() => {
         actions.loadAllModels(true);
     }, []);
 
+    // if (showAliquotViewSelector && activeSampleAliquotType != ALIQUOT_FILTER_MODE.all)
     const { queryModelsWithData, tabOrderWithData } = useMemo(() => {
         const models = {};
         const tabOrderWithData = tabOrder.slice();
-        Object.values(queryModels).forEach(model => {
+        let targetQueryModels = Object.values(queryModels);
+        const isFilteredView =
+            showAliquotViewSelector &&
+            activeSampleAliquotType != null &&
+            activeSampleAliquotType != ALIQUOT_FILTER_MODE.all;
+        if (isFilteredView) {
+            targetQueryModels = [];
+            Object.values(queryModels).forEach(model => {
+                if (model.id?.indexOf('unfiltered-assay-detail') === 0) targetQueryModels.push(model);
+            });
+        }
+
+        targetQueryModels.forEach(model => {
+            let targetModel = model;
+            if (isFilteredView) {
+                targetModel = Object.values(queryModels).find(m => m.id == model.id.substring('unfiltered-'.length));
+            }
             if (model.hasRows) {
-                models[model.id] = model;
+                models[targetModel.id] = targetModel;
             } else {
-                const idx = tabOrderWithData.findIndex(id => id === model.id);
+                const idx = tabOrderWithData.findIndex(id => id === targetModel.id);
                 if (idx > -1) {
                     tabOrderWithData.splice(idx, 1);
                 }
             }
         });
         return { queryModelsWithData: models, tabOrderWithData };
-    }, [allLoaded, queryModels]);
+    }, [allLoaded, queryModels, showAliquotViewSelector, activeSampleAliquotType]);
+
+    const getEmptyText = useCallback(
+        activeModel => {
+            if (!activeSampleAliquotType || activeSampleAliquotType == ALIQUOT_FILTER_MODE.all || activeModel.hasRows)
+                return undefined;
+
+            if (activeSampleAliquotType == ALIQUOT_FILTER_MODE.aliquots) {
+                return emptyAliquotViewMsg ?? 'No assay results available for aliquots of this sample.';
+            } else {
+                return (
+                    emptySampleViewMsg ??
+                    "Assay results are available for this sample's aliquots, but not available for this sample."
```

Comment:
How do we know that there are results for the aliquots at this point?

Reply:
If there is no assay data for either aliquots or the sample itself, then this assay design tab won't be displayed.